### PR TITLE
Create an MCG CLI pod to support macOS

### LINKED ISSUE #2477: mcg-cli causing OSError on Mac OSX executions
```
15:21:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: /Users/clacroix/workspace/ocs-ci/data/mcg-cli status -n openshift-storage
FAILED                                                                                                                                                                                                 [100%]
--------------------------------------------------------------------------------------------- live log teardown ----------------------------------------------------------------------------------------------
15:22:00 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc wait --for condition=ready pod -l app=rook-ceph-tools -n openshift-storage --timeout=120s
15:22:01 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage get pod -l 'app=rook-ceph-tools' -o jsonpath='{.items[0].metadata.name}'
15:22:01 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-storage exec rook-ceph-tools-56ffcd5fd7-pfsqm ceph health
15:22:03 - MainThread - ocs_ci.utility.utils - WARNING - Command stderr: kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl kubectl exec [POD] -- [COMMAND] instead.

15:22:03 - MainThread - ocs_ci.utility.utils - INFO - Ceph cluster health is HEALTH_OK.
15:22:03 - MainThread - tests.conftest - INFO - Ceph health check passed at teardown


================================================================================================== FAILURES ==================================================================================================
______________________________________________________________________________________________ test_deployment _______________________________________________________________________________________________

pvc_factory = <function pvc_factory_fixture.<locals>.factory at 0x12f6f4378>, pod_factory = <function pod_factory_fixture.<locals>.factory at 0x12f6f4268>

    @deployment
    @polarion_id(get_deployment_polarion_id())
    def test_deployment(pvc_factory, pod_factory):
        deploy = config.RUN['cli_params'].get('deploy')
        teardown = config.RUN['cli_params'].get('teardown')
        if not teardown or deploy:
            log.info("Verifying OCP cluster is running")
            assert is_cluster_running(config.ENV_DATA['cluster_path'])
            if not config.ENV_DATA['skip_ocs_deployment']:
                ocs_registry_image = config.DEPLOYMENT.get(
                    'ocs_registry_image'
                )
                ocs_install_verification(ocs_registry_image=ocs_registry_image)

                nb_eps = config.DEPLOYMENT.get('noobaa_endpoints')
                if nb_eps > 1:
                    change_noobaa_endpoints_count(nb_eps)

                # Check basic cluster functionality by creating resources
                # (pools, storageclasses, PVCs, pods - both CephFS and RBD),
                # run IO and delete the resources
                sanity_helpers = Sanity()
>               sanity_helpers.health_check()

tests/ecosystem/deployment/test_deployment.py:35:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/sanity_helpers.py:38: in health_check
    self.ceph_cluster.cluster_health_check(timeout=60)
ocs_ci/ocs/cluster.py:246: in cluster_health_check
    if not self.mcg_obj.status:
ocs_ci/ocs/resources/mcg.py:805: in status
    status = self.exec_mcg_cmd('status').stderr
ocs_ci/ocs/resources/mcg.py:737: in exec_mcg_cmd
    **kwargs
ocs_ci/utility/utils.py:446: in exec_cmd
    **kwargs
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py:472: in run
    with Popen(*popenargs, **kwargs) as process:
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py:775: in __init__
    restore_signals, start_new_session)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <subprocess.Popen object at 0x10a806898>, args = ['/Users/clacroix/workspace/ocs-ci/data/mcg-cli', 'status', '-n', 'openshift-storage'], executable = b'/Users/clacroix/workspace/ocs-ci/data/mcg-cli'
preexec_fn = None, close_fds = True, pass_fds = (), cwd = None, env = None, startupinfo = None, creationflags = 0, shell = False, p2cread = 16, p2cwrite = 18, c2pread = 19, c2pwrite = 20, errread = 21
errwrite = 22, restore_signals = True, start_new_session = False

    def _execute_child(self, args, executable, preexec_fn, close_fds,
                       pass_fds, cwd, env,
                       startupinfo, creationflags, shell,
                       p2cread, p2cwrite,
                       c2pread, c2pwrite,
                       errread, errwrite,
                       restore_signals, start_new_session):
        """Execute program (POSIX version)"""

        if isinstance(args, (str, bytes)):
            args = [args]
        else:
            args = list(args)

        if shell:
            # On Android the default shell is at '/system/bin/sh'.
            unix_shell = ('/system/bin/sh' if
                      hasattr(sys, 'getandroidapilevel') else '/bin/sh')
            args = [unix_shell, "-c"] + args
            if executable:
                args[0] = executable

        if executable is None:
            executable = args[0]
        orig_executable = executable

        # For transferring possible exec failure from child to parent.
        # Data format: "exception name:hex errno:description"
        # Pickle is not used; it is complex and involves memory allocation.
        errpipe_read, errpipe_write = os.pipe()
        # errpipe_write must not be in the standard io 0, 1, or 2 fd range.
        low_fds_to_close = []
        while errpipe_write < 3:
            low_fds_to_close.append(errpipe_write)
            errpipe_write = os.dup(errpipe_write)
        for low_fd in low_fds_to_close:
            os.close(low_fd)
        try:
            try:
                # We must avoid complex work that could involve
                # malloc or free in the child process to avoid
                # potential deadlocks, thus we do all this here.
                # and pass it to fork_exec()

                if env is not None:
                    env_list = []
                    for k, v in env.items():
                        k = os.fsencode(k)
                        if b'=' in k:
                            raise ValueError("illegal environment variable name")
                        env_list.append(k + b'=' + os.fsencode(v))
                else:
                    env_list = None  # Use execv instead of execve.
                executable = os.fsencode(executable)
                if os.path.dirname(executable):
                    executable_list = (executable,)
                else:
                    # This matches the behavior of os._execvpe().
                    executable_list = tuple(
                        os.path.join(os.fsencode(dir), executable)
                        for dir in os.get_exec_path(env))
                fds_to_keep = set(pass_fds)
                fds_to_keep.add(errpipe_write)
                self.pid = _posixsubprocess.fork_exec(
                        args, executable_list,
                        close_fds, tuple(sorted(map(int, fds_to_keep))),
                        cwd, env_list,
                        p2cread, p2cwrite, c2pread, c2pwrite,
                        errread, errwrite,
                        errpipe_read, errpipe_write,
                        restore_signals, start_new_session, preexec_fn)
                self._child_created = True
            finally:
                # be sure the FD is closed no matter what
                os.close(errpipe_write)

            # self._devnull is not always defined.
            devnull_fd = getattr(self, '_devnull', None)
            if p2cread != -1 and p2cwrite != -1 and p2cread != devnull_fd:
                os.close(p2cread)
            if c2pwrite != -1 and c2pread != -1 and c2pwrite != devnull_fd:
                os.close(c2pwrite)
            if errwrite != -1 and errread != -1 and errwrite != devnull_fd:
                os.close(errwrite)
            if devnull_fd is not None:
                os.close(devnull_fd)
            # Prevent a double close of these fds from __init__ on error.
            self._closed_child_pipe_fds = True

            # Wait for exec to fail or succeed; possibly raising an
            # exception (limited in size)
            errpipe_data = bytearray()
            while True:
                part = os.read(errpipe_read, 50000)
                errpipe_data += part
                if not part or len(errpipe_data) > 50000:
                    break
        finally:
            # be sure the FD is closed no matter what
            os.close(errpipe_read)

        if errpipe_data:
            try:
                pid, sts = os.waitpid(self.pid, 0)
                if pid == self.pid:
                    self._handle_exitstatus(sts)
                else:
                    self.returncode = sys.maxsize
            except ChildProcessError:
                pass

            try:
                exception_name, hex_errno, err_msg = (
                        errpipe_data.split(b':', 2))
                # The encoding here should match the encoding
                # written in by the subprocess implementations
                # like _posixsubprocess
                err_msg = err_msg.decode()
            except ValueError:
                exception_name = b'SubprocessError'
                hex_errno = b'0'
                err_msg = 'Bad exception data from child: {!r}'.format(
                              bytes(errpipe_data))
            child_exception_type = getattr(
                    builtins, exception_name.decode('ascii'),
                    SubprocessError)
            if issubclass(child_exception_type, OSError) and hex_errno:
                errno_num = int(hex_errno, 16)
                child_exec_never_called = (err_msg == "noexec")
                if child_exec_never_called:
                    err_msg = ""
                    # The error must be from chdir(cwd).
                    err_filename = cwd
                else:
                    err_filename = orig_executable
                if errno_num != 0:
                    err_msg = os.strerror(errno_num)
                    if errno_num == errno.ENOENT:
                        err_msg += ': ' + repr(err_filename)
>               raise child_exception_type(errno_num, err_msg, err_filename)
E               OSError: [Errno 8] Exec format error: '/Users/clacroix/workspace/ocs-ci/data/mcg-cli'

/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/subprocess.py:1522: OSError
```

### PR DESCRIPTION
**Pull request by MeridianExplorer**
Fixes #2477

### CODE REVIEW

File: ocs_ci/ocs/constants.py [hunk 1]
```diff
@@ -911,17 +923,19 @@
 RHEL_WORKERS_CONF = os.path.join(CONF_DIR, 'ocsci/aws_upi_rhel_workers.yaml')
 
 # Users
-NOOBAA_SERVICE_ACCOUNT = "system:serviceaccount:openshift-storage:noobaa"
```
Comment (MeridianExplorer) [inline ocs_ci/ocs/constants.py:914]: This wasn't used anywhere in the code since #2121

File: tests/conftest.py [hunk 2]
```diff
@@ -1578,16 +1582,110 @@ def rgw_obj_fixture(request):
...
+    ):
+        log.info('Downloading the NooBaa CLI tool from the operator pod, this might take a while.')
+        cmd = (
+            f"oc exec -n {config.ENV_DATA['cluster_namespace']} {nb_operator_pod.name}"
+            f" -- cat {constants.NOOBAA_OPERATOR_POD_CLI_PATH}"
+            f"> {constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH}"
+        )
+        subprocess.run(cmd, shell=True)
+        # Make sure the binary was copied properly and has the correct permissions
+        assert os.path.isfile(constants.NOOBAA_OPERATOR_LOCAL_CLI_PATH), (
```
Comment (vasukulkarni) [inline tests/conftest.py:1652]: should this assert and the one below be outside if statement?

Comment (MeridianExplorer) [reply tests/conftest.py:1652]: Those conditions are the same ones that are required to enter the this logic (see lines 1641-2). If they are true, the assertions will just re-check what's already checked. So I think that's a better place :)

Comment (vasukulkarni) [reply tests/conftest.py:1655]: the 'not' part is still confusing me, 1641-42 are entering using 'not' logic ?

Comment (vasukulkarni) [reply tests/conftest.py:1655]: specifically this case, which I think assert will be missed

```python
>>> (not True) or (not True)
False
```

**Pull request merged**